crates/krun/src/launch.rs

```diff
@@ -22,6 +23,29 @@ pub enum LaunchResult {
     },
 }
 
+#[derive(Debug)]
+enum LaunchError {
+    Connection(std::io::Error),
+    Json(serde_json::Error),
+    Server(String),
+}
+
+impl Display for LaunchError {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        match *self {
+            Self::Connection(ref err) => {
+                write!(f, "Can't connect to krun server: {err}")
+            },
+            Self::Json(ref err) => {
+                write!(f, "Can't transform arguments into JSON: {err}")
+            },
+            Self::Server(ref err) => {
+                write!(f, "krun server returned an error: {err}")
+            },
```

Comment:
```suggestion
            Self::Connection(ref err) => {
                write!(f, "could not connect to krun server: {err}")
            },
            Self::Json(ref err) => {
                write!(f, "could not serialize into JSON: {err}")
            },
            Self::Server(ref err) => {
                write!(f, "krun server returned an error: {err}")
            },
```

> Error messages are typically concise lowercase sentences without trailing punctuation

https://doc.rust-lang.org/std/error/trait.Error.html